### PR TITLE
2.0.0

### PR DESCRIPTION
**Pull request by ekmillard**
**SDK Enhancements Summary:**

**Type Annotations & Serializer Support:** Added detailed JMS Serializer annotations for all properties in Account, Dids, and Responses models to improve data handling.

**Error Handling Improvements:** Updated ErrorResponse model to include an optional statusCode, providing more robust error management across SDK calls.

**Method Refactoring:** Refined method names for clarity and consistency (e.g., renamed countryList2 to getRateZone2CountryList).

**Documentation Updates:** Added comprehensive PHPDoc comments to all classes and methods, ensuring clear and detailed documentation for future reference.

**Enhanced Query Handling:** Updated query calls to handle structured error responses and support for optional status codes.

**DidModel Updates:** Implemented fluent setters and added type annotations for all properties, ensuring stronger type safety.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,6 +10,11 @@
     },
     "minimum-stability": "dev",
     "require": {
-        "guzzlehttp/guzzle": "^7.9.2"
+        "guzzlehttp/guzzle": "^7.9.2",
+        "jms/serializer": "^3.31.1"
+    },
+    "require-dev": {
+        "phpunit/phpunit": "10.5.x-dev",
```
Comment (CoderJoeW) [inline composer.json]: We should update this to 11.4.3

File: composer.json [hunk 2]
```diff
@@ -10,6 +10,11 @@
...
+        "mockery/mockery": "2.0.x-dev"
```
Comment (CoderJoeW) [inline composer.json]: We should update this to 1.6.12

File: src/History.php [hunk 3]
```diff
@@ -2,35 +2,77 @@
...
+        ?string $period = null,
+        ?string $did = null,
+        ?string $type = null,
+        ?int $countryId = null,
+        ?bool $successCallOnly = null,
+        ?bool $summaryOnly = null,
+        int $page = 1,
+        int $perPage = 5,
+        ?string $timezone = null
+    ): string|ErrorResponse {
```
Comment (CoderJoeW) [inline src/History.php]: We should probably introduce a model so we dont have such a long parameter list

File: src/History.php [hunk 4]
```diff
@@ -2,35 +2,77 @@
...
             'type'              => $type,
             'country_id'        => $countryId,
-            'success_call_only' => $successCallOnly, // Zone 2 Call country ID
+            'success_call_only' => $successCallOnly,
             'summary_only'      => $summaryOnly,
             'page'              => $page,
-            'per_page'          => $perPage   // 5 - 500
+            'per_page'          => $perPage,
+            'timezone'          => $timezone,
         ];
```
Comment (CoderJoeW) [inline src/History.php]: Probably there is a better way then this perhaps we could have some object to array conversion that would reduce the duplication

File: src/Ifaxenterprise.php [hunk 5]
```diff
@@ -2,277 +2,449 @@
...
+     * @param string $faxLogin User login.
+     * @param string $faxPassword User password.
+     * @param string $sname Service name.
+     * @param string $faxName User's first name.
+     * @param string|null $faxLname User's last name.
+     * @param string|null $faxEmail User's email.
+     * @param string $isAdmin Flag to set as admin (default: 'off').
+     * @return string|ErrorResponse
+     */
+    public function createUser(string $faxLogin, string $faxPassword, string $sname, string $faxName, ?string $faxLname = null, ?string $faxEmail = null, string $isAdmin = 'off'): string|ErrorResponse
```
Comment (CoderJoeW) [inline src/Ifaxenterprise.php]: Huge parameter list here that could probably be replaced with a model

File: src/Ifaxenterprise.php [hunk 6]
```diff
@@ -2,277 +2,449 @@
...
+     *
+     * @param string $faxLogin User login.
+     * @param string $did The DID associated with the user.
+     * @param bool $allowSend Permission to allow sending.
+     * @param bool $allowDelete Permission to allow deletion.
+     * @param bool $allowListIn Permission to list incoming faxes.
+     * @param bool $allowListOut Permission to list outgoing faxes.
+     * @return string|ErrorResponse
+     */
+    public function updateUserPermissions(string $faxLogin, string $did, bool $allowSend, bool $allowDelete, bool $allowListIn, bool $allowListOut): string|ErrorResponse
```
Comment (CoderJoeW) [inline src/Ifaxenterprise.php]: Long parameter list

File: src/Ifaxenterprise.php [hunk 7]
```diff
@@ -2,277 +2,449 @@
...
+     * Retrieve fax history with optional filters.
+     *
+     * @param string|null $did Optional DID filter.
+     * @param string|null $service Optional service type filter.
+     * @param string|null $type Optional type filter.
+     * @param string|null $faxId Optional fax ID filter.
+     * @param string|null $period Optional period filter.
+     * @return string|ErrorResponse
+     */
+    public function faxHistory(?string $did = null, ?string $service = null, ?string $type = null, ?string $faxId = null, ?string $period = null): string|ErrorResponse
```
Comment (CoderJoeW) [inline src/Ifaxenterprise.php]: Long parameter list

File: src/Ifaxpro.php [hunk 8]
```diff
@@ -17,21 +36,62 @@ public function listRateCenters($state)
...
+     * @param string|null $pin Optional PIN for the DID.
+     * @param string $faxName The fax account name.
+     * @param string $faxEmail The email associated with the fax account.
+     * @param string $faxLogin The login for the fax account.
+     * @param string $faxPassword The password for the fax account.
+     * @param string|null $isFull Optional parameter indicating full feature mode.
+     * @param string|null $reportAtt Optional report attachment option.
+     * @return string|ErrorResponse|bool Returns true on success, or an error message on failure.
+     */
+    public function orderDid(
```
Comment (CoderJoeW) [inline src/Ifaxpro.php]: Long parameter list

File: composer.json (hunk 2 above)
Comment (ekmillard) [reply composer.json]: I agree with this.

File: composer.json (hunk 1 above)
Comment (ekmillard) [reply composer.json]: I agree with this, will update.

**Pull request closed without merging**